### PR TITLE
Daily Saved Export / Dashboard Feed redesign

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Apologies in advance for this monster... 😳

This PR reorganizes and adds functionality to the exports page. The changes are detailed in "phase 2" of [this document](https://docs.google.com/document/d/16svaBmyNiFGYEyaOMHKLUoJrLG9M7DjtwaNvrdusoH0/edit).

A summary of the major changes:
- move daily saved exports and excel dashboards to their own pages at the same level as "forms" and "cases"
- rename "excel dashboards" to "dashboard feeds"
- Allow users to save and edit filters for daily saved exports (previously impossible to do it for the *saved* export). This UI is in a modal on the export list page.

I tried to keep the commits of manageable size and a clean history, but there ends up being a bit of churn from e52b36b and on. I think all of the commits after that point are reconciling the changes from this ["locations export" PR](https://github.com/dimagi/commcare-hq/pull/13362) that happened simultaneously. Also given the shear number of commits it might be easier to review this all at once, not sure.

cc:
code buddy @biyeun 
initiative buddy @kaapstorm 
location exports expert @mkangia

### CODE REVIEW

**Comment by NoahCarnahan** [comment]
Adding tag because not sure if release needs to be timed with any comms

**Comment by dimagimon** [comment]
There are 33 errors:

* corehq/apps/export/static/export/js/list_exports.ng.js, line 90 - 'Clipboard' is not defined. (no-undef)
* corehq/apps/export/static/export/js/list_exports.ng.js, line 93 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/list_exports.ng.js, line 154 - 'oldSelectedExport' is defined but never used (no-unused-vars)
* corehq/apps/export/static/export/js/list_exports.ng.js, line 172 - 'oldDateRange' is defined but never used (no-unused-vars)
* corehq/apps/export/forms.py, line 107 - E501 line too long (124 > 115 characters)
* corehq/apps/export/forms.py, line 544 - E501 line too long (119 > 115 characters)
* corehq/apps/export/forms.py, line 563 - E501 line too long (116 > 115 characters)
* corehq/apps/export/forms.py, line 828 - E501 line too long (116 > 115 characters)
* corehq/apps/export/forms.py, line 947 - E501 line too long (136 > 115 characters)
* corehq/apps/export/forms.py, line 958 - E501 line too long (129 > 115 characters)
* corehq/apps/export/forms.py, line 983 - E501 line too long (119 > 115 characters)
* corehq/apps/export/forms.py, line 986 - E501 line too long (116 > 115 characters)
* corehq/apps/export/models/new.py, line 542 - E501 line too long (127 > 115 characters)
* corehq/apps/export/models/new.py, line 833 - E501 line too long (123 > 115 characters)
* corehq/apps/export/models/new.py, line 884 - E501 line too long (123 > 115 characters)
* corehq/apps/export/views.py, line 1060 - E501 line too long (125 > 115 characters)
* corehq/apps/export/views.py, line 1464 - E303 too many blank lines (2)
* corehq/apps/export/views.py, line 1889 - E501 line too long (124 > 115 characters)
* corehq/apps/reports/filters/case_list.py, line 3 - F401 'Group' imported but unused
* corehq/apps/reports/filters/case_list.py, line 4 - F401 'Location' imported but unused
* corehq/apps/reports/filters/case_list.py, line 5 - F401 'get_wrapped_owner' imported but unused
* corehq/apps/reports/filters/case_list.py, line 6 - F401 'CommCareUser' imported but unused
* corehq/apps/users/models.py, line 852 - E303 too many blank lines (2)
* corehq/tabs/tabclasses.py, line 604 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 605 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 606 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 607 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 625 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 636 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 637 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 638 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 639 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 657 - E122 continuation line missing indentation or outdented

**Comment by dimagimon** [comment]
There are 9 errors:

* corehq/apps/export/static/export/js/list_exports.ng.js, line 1 - 'clipboard' is defined but never used (no-unused-vars)
* corehq/apps/export/static/export/js/list_exports.ng.js, line 91 - 'Clipboard' is not defined. (no-undef)
* corehq/apps/export/forms.py, line 548 - E251 unexpected spaces around keyword / parameter equals
E502 the backslash is redundant between brackets
* corehq/apps/export/forms.py, line 549 - E131 continuation line unaligned for hanging indent
* corehq/tabs/tabclasses.py, line 636 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 637 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 638 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 639 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 657 - E122 continuation line missing indentation or outdented

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/export/forms.py, line 548 - E251 unexpected spaces around keyword / parameter equals
* corehq/apps/export/forms.py, line 549 - E131 continuation line unaligned for hanging indent
* corehq/tabs/tabclasses.py, line 636 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 637 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 638 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 639 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 657 - E122 continuation line missing indentation or outdented

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/export/forms.py, line 548 - E251 unexpected spaces around keyword / parameter equals
* corehq/apps/export/forms.py, line 549 - E131 continuation line unaligned for hanging indent
* corehq/tabs/tabclasses.py, line 636 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 637 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 638 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 639 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 657 - E122 continuation line missing indentation or outdented

File: corehq/apps/es/users.py
Comment (mkangia) [inline corehq/apps/es/users.py]: i am semi sure about this.
I think fetching locations without domain as in `user_ids_at_locations` would lead to more user_ids in the query than required for a particular domain if location_ids come straight from assigned_locations to a user which can span to different domains.
If this has already been considered when removing this method, great!

**Comment by mkangia** [comment]
WoW!! Thanks for taking time and moving code and making exports better @NoahCarnahan.. You the expert now :1st_place_medal:

File: corehq/apps/es/users.py
Comment (mkangia) [inline corehq/apps/es/users.py]: Noah, 
i have used the method `user_ids_at_accessible_locations` and hence `users_at_accessible_locations` in UserES in couple of PRs and are now on staging. Can you put these back so they are not removed on staging, unless there is a strong reason to remove them or if there is an alternate?

File: corehq/apps/export/templates/export/customize_export_new.html
Comment (benrudolph) [inline corehq/apps/export/templates/export/customize_export_new.html]: i think this template only gets used if it's use_new_exports. otherwise the `customize_export_old` gets loaded

File: corehq/apps/export/tests/test_export_views.py
Comment (benrudolph) [inline corehq/apps/export/tests/test_export_views.py]: nice view tests

File: corehq/apps/export/urls.py
Comment (benrudolph) [inline corehq/apps/export/urls.py]: 🙌 for breaking up these urls

**Review by benrudolph — COMMENTED** [review]
awesome!! didn't review super super closely, but very excited for these changes

File: corehq/apps/es/users.py
Comment (NoahCarnahan) [inline corehq/apps/es/users.py]: Hey, thanks for pointing this out. I just checked on it, and when I use `user_ids_at_locations()` in the scope filter, the locations don't come straight from a user's assigned locations, I use `SQLLocation.active_objects.accessible_location_ids(domain, couch_user))` to get the locations, which is the same thing that `UserES.users_at_accessible_locations()` does, so I think this should be safe.

I'll add back the `user_ids_at_accessible_locations()` function here to avoid the issues with the other branches on staging.

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/export/forms.py, line 548 - E251 unexpected spaces around keyword / parameter equals
* corehq/apps/export/forms.py, line 549 - E131 continuation line unaligned for hanging indent
* corehq/tabs/tabclasses.py, line 636 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 637 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 638 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 639 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 657 - E122 continuation line missing indentation or outdented

File: corehq/apps/export/templates/export/customize_export_new.html
Comment (NoahCarnahan) [inline corehq/apps/export/templates/export/customize_export_new.html]: good catch, thanks. Fixed.

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/export/forms.py, line 548 - E251 unexpected spaces around keyword / parameter equals
* corehq/apps/export/forms.py, line 549 - E131 continuation line unaligned for hanging indent
* corehq/tabs/tabclasses.py, line 636 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 637 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 638 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 639 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 657 - E122 continuation line missing indentation or outdented

**Comment by mkangia** [comment]
@NoahCarnahan did one [quick commit](https://github.com/dimagi/commcare-hq/pull/14440/commits/83ad01d8d2f7142b99751522718e9932939b66f2) to reflect a change in submit history and manage forms.

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/export/forms.py, line 548 - E251 unexpected spaces around keyword / parameter equals
* corehq/apps/export/forms.py, line 549 - E131 continuation line unaligned for hanging indent
* corehq/tabs/tabclasses.py, line 636 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 637 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 638 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 639 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 657 - E122 continuation line missing indentation or outdented

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/export/forms.py, line 548 - E251 unexpected spaces around keyword / parameter equals
* corehq/apps/export/forms.py, line 549 - E131 continuation line unaligned for hanging indent
* corehq/tabs/tabclasses.py, line 636 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 637 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 638 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 639 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 657 - E122 continuation line missing indentation or outdented

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/export/forms.py, line 548 - E251 unexpected spaces around keyword / parameter equals
* corehq/apps/export/forms.py, line 549 - E131 continuation line unaligned for hanging indent
* corehq/apps/reports/filters/users.py, line 8 - F401 'Location' imported but unused
* corehq/tabs/tabclasses.py, line 636 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 637 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 638 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 639 - E122 continuation line missing indentation or outdented
* corehq/tabs/tabclasses.py, line 657 - E122 continuation line missing indentation or outdented

**Comment by NoahCarnahan** [comment]
adding "hold off on merge" tag because product raised some questions about if documentation should be updated before this is deployed.

File: corehq/apps/export/forms.py
Comment (dimagimon) [inline corehq/apps/export/forms.py]: E251 unexpected spaces around keyword / parameter equals

File: corehq/apps/export/forms.py
Comment (dimagimon) [inline corehq/apps/export/forms.py]: E131 continuation line unaligned for hanging indent

File: corehq/tabs/tabclasses.py
Comment (dimagimon) [inline corehq/tabs/tabclasses.py]: E122 continuation line missing indentation or outdented

**Comment by NoahCarnahan** [comment]
Hey @benrudolph and @mkangia, thanks for the review you did on this PR earlier. I think neither of you "approved" it (correct me if I'm wrong, new github UI is a little confusing). Were you looking for any changes, or did you just feel like you hadn't gone over it thoroughly enough? I'm discussing docs with Sheel right now, but once that is resolved I'm eager to have this merged soon to avoid issues that could come up with related code being changed on more recent branches.
Let me know if you need any assistance reviewing or if you're too busy, I'm happy to pull in my current code buddy.

**Comment by benrudolph** [comment]
hey @NoahCarnahan i had just done a quick review (was just interested when i saw the title of the PR), but i'll do a more thorough one tomorrow and approve/merge

File: corehq/apps/es/tests/test_esquery.py
Comment (benrudolph) [inline corehq/apps/es/tests/test_esquery.py]: seems odd that this test was deleted

File: corehq/apps/export/models/new.py
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: nit: user_ids

File: corehq/apps/export/models/new.py
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: may want to comment as to why `accessible_location_ids` is needed when we have  `locations`

File: corehq/apps/export/models/new.py
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: nit: the return type of an array and the pluralized `filters` makes me think that this is an array object when it's not

File: corehq/apps/export/models/new.py
Comment (benrudolph) [inline corehq/apps/export/models/new.py]: this is kind of magical to me. why can't the default value of user_types by `[0]`. also a constant instead of 0 would be helpful

File: corehq/apps/export/static/export/js/list_exports.ng.js
Comment (benrudolph) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: nice i love this library

File: corehq/apps/export/static/export/js/list_exports.ng.js
Comment (benrudolph) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: I can't find where this is used, but you can use momentjs to do more robust date checking: https://momentjs.com/docs/#/parsing/string-format/

File: corehq/apps/export/static/export/js/list_exports.ng.js
Comment (benrudolph) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: nit: i often like to do
```
if (!newSelectedExport) {
return;
}
```
so there's one less layer nesting

File: corehq/apps/export/templates/export/partial/feed_filter_modal.html
Comment (benrudolph) [inline corehq/apps/export/templates/export/partial/feed_filter_modal.html]: nit: trans

File: corehq/apps/export/templates/export/partial/feed_filter_modal.html
Comment (benrudolph) [inline corehq/apps/export/templates/export/partial/feed_filter_modal.html]: trans

File: corehq/tabs/tabclasses.py
Comment (benrudolph) [inline corehq/tabs/tabclasses.py]: is this else statement supposed to be tabbed over one? it seems like it should be "if you can view daily saved, then view the new ones if you're able else view the old". this seems to be saying "if you can't view daily saved exports, show you the old view"

File: corehq/tabs/tabclasses.py
Comment (benrudolph) [inline corehq/tabs/tabclasses.py]: same here

**Review by benrudolph — APPROVED** [review]
Looking good! nothing blocking as long as that daily saveds show up correctly for those who cannot view daily saved is correct. feel free to merge and follow up on whatever you deem fit

**Comment by benrudolph** [comment]
also tag is still on

**Comment by NoahCarnahan** [comment]
Thank you very much for the review @benrudolph! I'll push some changes tomorrow.

File: corehq/apps/export/forms.py
Comment (dimagimon) [inline corehq/apps/export/forms.py]: W293 blank line contains whitespace

File: corehq/tabs/tabclasses.py
Comment (dimagimon) [inline corehq/tabs/tabclasses.py]: E211 whitespace before '('

File: corehq/apps/export/static/export/js/list_exports.ng.js
Comment (NoahCarnahan) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: I'm going to punt this for now if you don't mind. I'm replicating the behavior of the saved reports date inputs, but I think the best solution is actually just replace these text fields with date picking widgets. That will take a little bit of time though and I'm trying to wrap this up... feel free to push back though if you think its important :)

Comment (benrudolph) [inline corehq/apps/export/static/export/js/list_exports.ng.js]: 👍 don't think it's that important

**Comment by benrudolph** [comment]
sweet looks great! thanks for the followup

test failures look real:
```
======================================================================
ERROR: test_edit_daily_saved_export_filters (corehq.apps.export.tests.test_export_views.ExportViewTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/mnt/commcare-hq-ro/corehq/apps/export/tests/test_export_views.py", line 204, in test_edit_daily_saved_export_filters
    self.assertEqual(export.filters.date_period.period_type, 'range')
AttributeError: 'NoneType' object has no attribute 'period_type'
-------------------- >> begin captured logging << --------------------
```

File: corehq/apps/export/forms.py
Comment (dimagimon) [inline corehq/apps/export/forms.py]: E501 line too long (121 > 115 characters)

File: corehq/apps/export/tests/test_export_instance.py
Comment (dimagimon) [inline corehq/apps/export/tests/test_export_instance.py]: F401 'FormExportInstanceFilters' imported but unused

**Comment by benrudolph** [comment]
oh baby! 💥 💥

**Pull request merged**
